data/src/main/java/org/apache/iceberg/data/ContentFileWriteBuilderImpl.java

```diff
@@ -0,0 +1,251 @@
+    properties.forEach(writeBuilder::set);
+    return (C) this;
+  }
+
+  @Override
+  public C meta(String property, String value) {
+    writeBuilder.meta(property, value);
+    return (C) this;
+  }
+
+  @Override
+  public C meta(Map<String, String> properties) {
+    properties.forEach(writeBuilder::meta);
+    return (C) this;
+  }
+
+  @Override
+  public C metricsConfig(MetricsConfig newMetricsConfig) {
+    writeBuilder.metricsConfig(newMetricsConfig);
+    return (C) this;
+  }
+
+  @Override
+  public C overwrite() {
+    writeBuilder.overwrite();
+    return (C) this;
+  }
+
+  @Override
+  public C fileEncryptionKey(ByteBuffer encryptionKey) {
+    writeBuilder.fileEncryptionKey(encryptionKey);
+    return (C) this;
+  }
+
+  @Override
+  public C aadPrefix(ByteBuffer aadPrefix) {
+    writeBuilder.aadPrefix(aadPrefix);
+    return (C) this;
+  }
+
+  @Override
+  public C rowSchema(Schema newSchema) {
+    this.rowSchema = newSchema;
+    return (C) this;
+  }
+
+  @Override
+  public C equalityFieldIds(List<Integer> fieldIds) {
+    this.equalityFieldIds = ArrayUtil.toIntArray(fieldIds);
+    return (C) this;
+  }
+
+  @Override
+  public C equalityFieldIds(int... fieldIds) {
+    this.equalityFieldIds = fieldIds;
+    return (C) this;
+  }
+
+  @Override
+  public C spec(PartitionSpec newSpec) {
+    this.spec = newSpec;
+    return (C) this;
+  }
+
+  @Override
+  public C partition(StructLike newPartition) {
+    this.partition = newPartition;
+    return (C) this;
+  }
+
+  @Override
+  public C keyMetadata(EncryptionKeyMetadata metadata) {
+    this.keyMetadata = metadata;
+    return (C) this;
+  }
+
+  @Override
+  public C sortOrder(SortOrder newSortOrder) {
+    this.sortOrder = newSortOrder;
+    return (C) this;
+  }
+
+  @Override
+  public <D> DataWriter<D> dataWriter() throws IOException {
+    Preconditions.checkArgument(spec != null, "Cannot create data writer without spec");
+    Preconditions.checkArgument(
+        spec.isUnpartitioned() || partition != null,
+        "Partition must not be null when creating data writer for partitioned spec");
+
+    return new DataWriter<>(
+        writeBuilder.build(), format, location, spec, partition, keyMetadata, sortOrder);
+  }
+
+  @Override
+  public <D> EqualityDeleteWriter<D> equalityDeleteWriter() throws IOException {
+    Preconditions.checkState(
+        rowSchema != null, "Cannot create equality delete file without a schema");
+    Preconditions.checkState(
+        equalityFieldIds != null, "Cannot create equality delete file without delete field ids");
+    Preconditions.checkArgument(
+        spec != null, "Spec must not be null when creating equality delete writer");
+    Preconditions.checkArgument(
+        spec.isUnpartitioned() || partition != null,
+        "Partition must not be null for partitioned writes");
+
+    return new EqualityDeleteWriter<>(
+        writeBuilder
+            .schema(rowSchema)
+            .meta("delete-type", "equality")
```

Comment:
I know this is the practice of existing code. but these constants have been repeated in multiple classes (for each file format). should we define some constants in the `iceberg-data` module?

Reply:
This will be the only place for these constants after the refactor and deprecation.
So I would keep them as it is for now.